### ISSUE TITLE
How exactly does dtaidistance keep track of labels?

### ISSUE THREAD

**Issue by ApoG22** [wannesm/dtaidistance#35]
Hi there! I am using your library for my master's thesis. Well, at least I am trying to. I almost have it done, I browsed the documentation, the source code and the closed issues - but I can't seem to find the solution to these three problems.

1. How does your library keep track of the labels??

I want to make dead sure the labels are matching the time series ID that I have. That's because the time series have pre-defined groups and I am checking whether the groups match the clusters. I did some manual checks, and it looks like I get it, but since I am posting anyway figured it's better to make double sure.

2. Is there a possibility to color the time series line charts by label? 

Let's say I have 1000 time series and they all have a label: A, B or C. I would like A series to be red, B to be blue, C to be green.

3. Can I extract a list of labels at each cluster?

I would like to know which time series are grouped together, and at what level of clustering certain attributes of my time-series make them cluster together.

Here is my code:
````
#Thats the database, the values I am clustering on is a floating point in NDVI column
database = pd.read_excel(r"data.xlsx")
data=database[['NDVI', 'Data', 'ID', 'TextAttribute']].dropna()
data.sort_values(by=['ID','Data'], ascending=True)

#Labels
duplicates = data.drop_duplicates(subset=['ID','TextAttribute'])
labels = duplicates['PartID'].tolist()

values = data.groupby('ID')['NDVI'].apply(lambda x: x.to_numpy())
series = [x.astype(np.double) for x in L]

#The clustering
model = clustering.LinkageTree(dtw.distance_matrix_fast, {})
cluster_idx = model.fit(series)

#Plotting
fig, ax = plt.subplots(nrows=1, ncols=2, figsize=(20, 250), dpi=300)
model.plot(filename='rawdata_all_dtwvis_id.pdf', axes=ax, show_ts_label=labels,
           show_tr_label=True, ts_label_margin=-35,
           ts_left_margin=30)

````

**Comment by wannesm** [comment]
There are no labels. The toolbox only uses the indices, thus the index of the series in the list of series.
The plot function that is available can print a custom label by passing to `show_ts_label` a function that takes the index as an argument and returns a string.

The built in cluster function does not return cluster labels or groups. It returns the linkage data structure as used in SciPy ( https://docs.scipy.org/doc/scipy/reference/generated/scipy.cluster.hierarchy.linkage.html ). The idea is that once the linkage is available you can use the SciPy methods to go further. For example https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.cluster.hierarchy.fcluster.html 

If you want to use the result of fcluster or another method in the plot function, I added the `ts_color` argument to return the color for each series (again based on index).
